baseplate/__init__.py

```diff
@@ -1,215 +1,624 @@
-import os
-import sys
+import logging
+import random
 
-from typing import TYPE_CHECKING
+from types import TracebackType
+from typing import Tuple, Optional, Type, NamedTuple, Any, Dict
 
-from baseplate import config, metrics
-from baseplate.core import Baseplate
-from baseplate.diagnostics import tracing
-from baseplate._utils import warn_deprecated
+from baseplate.lib import config, warn_deprecated
 
 
-if TYPE_CHECKING:
-    import raven
+logger = logging.getLogger(__name__)
 
 
-def metrics_client_from_config(raw_config: config.RawConfig) -> metrics.Client:
-    """Configure and return a metrics client.
+class BaseplateObserver:
+    """Interface for an observer that watches Baseplate."""
 
-    This expects two configuration options:
+    def on_server_span_created(self, context: Any, server_span: "Span") -> None:
```

Comment:
Shouldn't we be using the `RequestContext` type here instead of `Any`?

Reply:
Probably, now that it exists. I'm going to do another pass of typing after this is merged and will make sure to fix that up. GP!